### PR TITLE
Process and validate id token

### LINKED ISSUE #80: validateIdToken
Required:
- iss
- sub
- aud
- exp
- iat

Required if specified in config:
- nonce
- auth_time (if max_age is specified)

Optional attributes to validate:
- acr
- amr
- azp

Look into additional validation of the jwt header.

### LINKED ISSUE #81: Collect and store id_token in token request of OpenIDClient

### PR DESCRIPTION
**Pull request by perarnborg**
Closes #80
Closes #81 

Also in this PR:
- Configure dev to run against the demo issuer https://demo.duendesoftware.com
- Make dev authentication flow work

Real e2e tests of authentication still on hold until we have our own provider to use.

### CODE REVIEW

File: src/createStorageModule.ts [hunk 1]
```diff
@@ -37,7 +37,12 @@ export const createStorageModule = (
   }
 
   const _hashedKey = (key: StorageKey): string => {
-    return btoa(`${config.issuer}-${config.clientId}-${key}`)
+    console.log((config.scopes || []).join('_'))
```
Comment (dennerkrans) [inline src/createStorageModule.ts:40]: Do we need this console.log?

**Review by dennerkrans — APPROVED** [review]
Nice 👍 Added a comment about a console.log that might've snuck in. If it should be there merge away :)

**Comment by perarnborg** [comment]
@dennerkrans Thanks! It’s gone!

**Review by dennerkrans — APPROVED** [review]
🎊

**Pull request merged**